dashboard/test/models/pd/application/application_base_test.rb

```diff
@@ -8,6 +8,10 @@ class ApplicationBaseTest < ActiveSupport::TestCase
 
     freeze_time
 
+    setup_all do
+      Pd::Application::ApplicationBase.any_instance.stubs(:deliver_email)
+    end
```

Comment:
do you have an understanding of what happens to this stub after this test case runs (and finishes) later in this file? https://github.com/code-dot-org/code-dot-org/blob/2650e3e42b51d0635b1b9b2f2d2d324887b03cca/dashboard/test/models/pd/application/application_base_test.rb#L273

for example, when that test case finishes, do we go back to actually calling `deliver_email` in subsequent test cases, or does the original stub from `setup_all` get restored?

I'm not sure I could say for sure from looking at it. one idea (as an alternative to investigating) would be to use `setup` instead of `setup_all` here.

Reply:
Oh great question –– I'll investigate

Reply:
I looked into trying to do an expect of a method that runs before the `deliver_email` stub, but nothing really made sense. So I went ahead and change the `setup_all` to `setup` as you recommended